parser-typechecker/src/Unison/Builtin.hs

```diff
@@ -503,6 +506,22 @@ mvarBuiltins =
   mvar :: Type v -> Type v
   mvar a = Type.ref () Type.mvarRef `app` a
 
+codeBuiltins :: forall v. Var v => [(Text, Type v)]
+codeBuiltins =
+  [ ("Code.dependencies", code --> list termLink)
+  , ("Code.isMissing", termLink --> io boolean)
+  , ("Code.serialize", code --> bytes)
+  , ("Code.deserialize", bytes --> eithert text code)
+  , ("Code.cache_", list (pair termLink code) --> io (list termLink))
```

Comment:
The definition should have type `[(Link.Term,Code)] ->{IO} [Link.Term]`. Tuples are right-nested pairs terminated by a `()`.

Just use this handy helper function in `Builtin.hs`. It's used by `Text.uncons` if you want to see an example. 

```suggestion
  , ("Code.cache_", list (tuple [termLink, code]) --> io (list termLink))
```

```Haskell
tuple :: Ord v => [Type v] -> Type v
tuple [t] = t
tuple ts = foldr pair (DD.unitType ()) ts
```